R/plot-individual-time-profile.R

```diff
@@ -0,0 +1,136 @@
+#' Concentration time profile plot for individual
+#'
+#' @param dataCombined A `DataCombined` object.
+#' @param defaultPlotConfiguration A `DefaultPlotConfiguration` object, which is
+#'   an `R6` class object that defines plot properties.
+#'
+#' @import tlf
+#'
+#' @family plotting
+#'
+#' @examples
+#'
+#' # TODO: add example
+#'
+#' @export
+plotIndividualTimeProfile <- function(dataCombined,
+                                      defaultPlotConfiguration = NULL) {
+  # validation -----------------------------
+
+  defaultPlotConfiguration <- defaultPlotConfiguration %||% DefaultPlotConfiguration$new()
+  validateIsOfType(dataCombined, "DataCombined")
+  validateIsSameLength(objectCount(dataCombined), 1L) # only single instance is allowed
+  validateIsOfType(defaultPlotConfiguration, "DefaultPlotConfiguration", nullAllowed = FALSE)
+
+  # data frames -----------------------------
+
+  df <- dataCombined$toDataFrame()
+
+  # Getting all units on the same scale
+  df <- .unitConverter(df, defaultPlotConfiguration$xUnit, defaultPlotConfiguration$yUnit)
+
+  # Datasets which haven't been assigned to any group will be plotted as a group
+  # on its own. That is, the `group` column entries for them will be their names.
+  df <- .addMissingGroupings(df)
+
+  # TimeProfilePlotConfiguration object -----------------------------
+
+  # Create an instance of `defaultInternalPlotConfiguration` class
+  defaultInternalPlotConfiguration <- .createDefaultInternalPlotConfiguration(defaultPlotConfiguration)
+
+  # Create an instance of `TimeProfilePlotConfiguration` class
+  individualTimeProfilePlotConfiguration <- tlf::TimeProfilePlotConfiguration$new()
+
+  individualTimeProfilePlotConfiguration$labels <- defaultInternalPlotConfiguration$labels
```

Comment:
It feels like this should be a method that takes a tlf object , a defaultPlotConfiguration, and creates the new internal plot config and map it. I am pretty sure I am going to see the same code in all different plot types....am I wrong?